index.js

```diff
@@ -1,15 +1,19 @@
 "use strict";
 
+function initBuffer(val) {
+    return Buffer.from(val);
+}
+
 if (!Buffer.prototype.indexOf) {
     Buffer.prototype.indexOf = function (value, offset) {
         offset = offset || 0;
 
         // Always wrap the input as a Buffer so that this method will support any
         // data type such as array octet, string or buffer.
         if (typeof value === "string" || value instanceof String) {
-            value = new Buffer(value);
+            value = initBuffer(value);
```

Comment:
Can you simply call Buffer.form(value) here and below? I don't see a strong reason to create initBuffer to just wrap around Buffer.from.

Reply:
Done.